src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,47 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int ARRAY_SIZE = 10;
+    private static final int FIRST_INDEX = 0;
+    private static final int DECREMENT = 1;
```

Comment:
I think that these would be self-explanatory written directly in code, without creating constants

Reply:
++

Reply:
done